esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,486 @@
+        logger.info("input_dir  = %s", self.rawobs)
+        # check if the inputdir actually exists
+        if not os.path.isdir(self.rawobs):
+            logger.error("Directory %s does not exist", self.rawobs)
+            raise ValueError
+        logger.info("output_dir = %s", self.output_dir)
+        logger.info(70 * "-")
+
+    @property
+    def rawobs(self):
+        """Raw obs folder path."""
+        return self.config["rootpath"]["RAWOBS"][0]
+
+    @property
+    def output_dir(self):
+        """Output folder path."""
+        return self.config['output_dir']
+
+    @property
+    def run_dir(self):
+        """Run dir folder path."""
+        return os.path.join(self.config['output_dir'], 'run')
+
+    @property
+    def log_level(self):
+        """"Console log level."""
+        return self.config['log_level']
+
+    @staticmethod
+    def _dataset_to_module(dataset):
+        return dataset.lower().replace('-', '_')
+
+    def download(self, start_date, end_date, overwrite):
+        """Download all datasets.
+
+        Parameters
+        ----------
+        start_date: datetime
+            First date to download
+        end_date: datetime
+            Last date to download
+        overwrite: boolean
+            If True, download again existing files
+        """
+        if not self.datasets:
+            logger.error('Missing datasets to download')
+        logger.info("Downloading original data...")
+        # master directory
+        failed_datasets = []
+        for dataset in self.datasets:
+            try:
+                self.download_dataset(dataset, start_date, end_date, overwrite)
+            except Exception:
+                logger.exception('Failed to download %s', dataset)
+                failed_datasets.append(dataset)
+        if failed_datasets:
+            logger.error('Download failed for datasets %s', self.datasets)
+
+    def download_dataset(self, dataset, start_date, end_date, overwrite):
+        """Download a single dataset.
+
+        Parameters
+        ----------
+        dataset: str
+            Dataset name
+        start_date: datetime
+            First date to download
+        end_date: datetime
+            Last date to download
+        overwrite: boolean
+            If True, download again existing files
+        """
+        dataset_module = self._dataset_to_module(dataset)
+        logger.info('Downloading %s', dataset)
+        logger.debug("Download module: %s", dataset_module)
+        try:
+            downloader = importlib.import_module(
+                f'.{dataset_module}',
+                package='esmvaltool.cmorizers.data.downloaders.datasets')
+        except ImportError:
+            logger.exception('Could not find cmorizer for %s', dataset)
+            raise
+
+        downloader.download_dataset(self.config, dataset, start_date, end_date,
+                                    overwrite)
+        logger.info('%s downloaded', dataset)
+
```

Comment:
In some cases, running several times the command `esmvaltool data download --overwrite=True DATASET` produces several copies of the same files which are named "filename.nc", "filename.nc.1", "filename.nc.2", ... Examples are `PHC`, `HadCRUT3`, `HadCRUT4`, `Duveiller2018`. But some over cases work fine. I'm not sure what is the reason behing that.

Reply:
All of them are using wget, so probably is an issue with that particular downloader.

I will take a look

Reply:
This is a limitation with wget. You must provide the filename if you want it to really overwrite the folder.

What I have done to keep it simple is to do it for downloading specific files and raise an error if we are trying to download a folder and use the overwrite flag.

I think is enough, as it is just a matter of removing the data manually, but if that becomes a problem we can complicate the code to achieve this later on

Reply:
Works better now, thanks for the changes. The only thing I noted is that when downloading for a second time a dataset (e.g. `CowtanWay`) using `esmvaltool data download --overwrite=True CowtanWay`, the tarball *.nc.gz remains in the current directory which is not the case with the first download.

Reply:
In fact, whenever the `--overwrite=True` option is used for the wget-based downloader, the old files are not overwritten. Instead, new files are donwlowded in the working directory. I understand this is a limitation of wget and there is no easy solution to that. Should we display a warning when the overwrite option is used with wget-based downloader to inform users that old files should be manually deleted first or simply to check their working directory? There is a (slight) risk that a users fills the disk space quickly if such commands are run from their $HOME directory.